docs/DeveloperGuide.md

```diff
@@ -460,8 +461,10 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **CLI**: A text-based user interface used to run programs
+* **GUI**: A graphical user interface (GUI) is a form of user interface that allows users to interact with programs through graphical icons and audio indicator
 * **Private contact detail**: A contact detail that is not meant to be shared with others
-
+* **Autocomplete**: A feature that shows a list of completed words or strings without the user needing to type them in full
 --------------------------------------------------------------------------------------------------------------------
```

Comment:
Should we include terms like `SoConnect`, `NUS SoC`, `JavaFX`, `kLoC`, etc.?

Reply:
`SoConnect` is the name of our product and it is explained in the introduction, so I think we do not have to repeat it again in the glossary, but the rest I think can include them. Thanks for the suggestion!